app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -81,14 +83,14 @@ class Bolt_Boltpay_Model_Payment extends Mage_Payment_Model_Method_Abstract
 
     protected $_validStateTransitions = array(
         self::TRANSACTION_AUTHORIZED => array(self::TRANSACTION_COMPLETED, self::TRANSACTION_CANCELLED, self::TRANSACTION_REJECTED_REVERSIBLE, self::TRANSACTION_REJECTED_IRREVERSIBLE),
-        self::TRANSACTION_COMPLETED => array(self::TRANSACTION_NO_NEW_STATE),
+        self::TRANSACTION_COMPLETED => array(self::TRANSACTION_REFUND, self::TRANSACTION_NO_NEW_STATE),
         self::TRANSACTION_PENDING => array(self::TRANSACTION_AUTHORIZED, self::TRANSACTION_CANCELLED, self::TRANSACTION_REJECTED_REVERSIBLE, self::TRANSACTION_REJECTED_IRREVERSIBLE, self::TRANSACTION_COMPLETED),
         self::TRANSACTION_REJECTED_IRREVERSIBLE => array(self::TRANSACTION_NO_NEW_STATE),
         self::TRANSACTION_REJECTED_REVERSIBLE => array(self::TRANSACTION_AUTHORIZED, self::TRANSACTION_CANCELLED, self::TRANSACTION_REJECTED_IRREVERSIBLE, self::TRANSACTION_COMPLETED),
-        self::TRANSACTION_CANCELLED => array(self::TRANSACTION_NO_NEW_STATE)
+        self::TRANSACTION_CANCELLED => array(self::TRANSACTION_NO_NEW_STATE),
+        self::TRANSACTION_REFUND => array(self::TRANSACTION_REFUND,self::TRANSACTION_NO_NEW_STATE)
```

Comment:
I don't think you have to define self::TRANSACTION_REFUND -> self::TRANSACTION_REFUND

Reply:
well, as I explained, this array is defined for next valid status, then for partial refund, we would refund for several times and the status would keep TRANSACTION_REFUND  then

Reply:
```
$validNextStatuses = null;
                if (array_key_exists($prevTransactionStatus, $this->_validStateTransitions)) {
                    $validNextStatuses = $this->_validStateTransitions[$prevTransactionStatus];
                } else {
                    $message = sprintf("Invalid previous state: %s", $prevTransactionStatus);
                    Mage::throwException($message);
                }
```

Reply:
@dumega What you say is true.  Using the sentinel value could avoid this in the exception throwing logic, but consecutive refunds if commented, is just as clear.